int64 movement

---
int64 check
int64 x_pixel
int64 y_pixel